waspc/cli/src/Wasp/Cli/Command/CreateNewProject.hs

```diff
@@ -26,99 +26,70 @@ data ProjectInfo = ProjectInfo
+        ]
   where
     appName = kebabToCamelCase name
 
-createNewProject :: String -> Command ()
-createNewProject name =
-  case parseProjectInfo name of
-    Right projectInfo -> createNewProject' projectInfo
-    Left parsedError ->
-      throwError $
-        CommandError "Project creation failed" parsedError
-
-createNewProject' :: ProjectInfo -> Command ()
-createNewProject' (ProjectInfo projectName appName) = do
+createWaspProjectDir :: ProjectInfo -> Command ()
+createWaspProjectDir projectInfo = do
+  absWaspProjectDir <- getAbsoluteWaspProjectDir projectInfo
+  dirExists <- doesDirExist $ toPathAbsDir absWaspProjectDir
+  if dirExists
+    then throwProjectCreationError $ show absWaspProjectDir ++ " is an existing directory"
+    else liftIO $ do
+      initializeProjectFromSkeleton absWaspProjectDir
+      writeMainWaspFile absWaspProjectDir projectInfo
+
+getAbsoluteWaspProjectDir :: ProjectInfo -> Command (Path System Abs (Dir WaspProjectDir))
+getAbsoluteWaspProjectDir (ProjectInfo projectName _) = do
   absCwd <- liftIO getCurrentDirectory
-  waspProjectDir <- case SP.parseAbsDir $ absCwd FP.</> projectName of
-    Left err ->
-      throwError $
-        CommandError
-          "Project creation failed"
-          ( "Failed to parse absolute path to wasp project dir: "
-              ++ show err
-          )
+  case parseAbsDir $ absCwd FP.</> projectName of
     Right sp -> return sp
-  liftIO $ do
-    createDirectorySP waspProjectDir
-    writeFileSP (waspProjectDir </> mainWaspFileInWaspProjectDir) mainWaspFileContent
-    writeFileSP (waspProjectDir </> gitignoreFileInWaspProjectDir) gitignoreFileContent
-    writeFileSP
-      (waspProjectDir </> Common.dotWaspRootFileInWaspProjectDir)
-      "File marking the root of Wasp project."
-
-  let extCodeDir = waspProjectDir </> Common.extCodeDirInWaspProjectDir
-  liftIO $ do
-    createDirectorySP extCodeDir
-    dataDir <- Wasp.Data.getAbsDataDirPath
-
-    let copyTemplateFile' = copyTemplateFile dataDir extCodeDir
-
-    writeFileSP (extCodeDir </> waspignoreFileInExtCodeDir) waspignoreFileContent
-
-    copyTemplateFile'
-      [relfile|new/ext/MainPage.js|]
-      mainPageJsFileInExtCodeDir
-
-    copyTemplateFile'
-      [relfile|new/ext/Main.css|]
-      mainCssFileInExtCodeDir
-
-    copyTemplateFile'
-      [relfile|new/ext/waspLogo.png|]
-      waspLogoFileInExtCodeDir
-
-  liftIO $ do
-    putStrLn $ Term.applyStyles [Term.Green] ("Created new Wasp app in ./" ++ projectName ++ " directory!")
-    putStrLn "To run it, do:"
-    putStrLn ""
-    putStrLn $ Term.applyStyles [Term.Bold] ("    cd " ++ projectName)
-    putStrLn $ Term.applyStyles [Term.Bold] "    wasp start"
-    putStrLn ""
-    putStrLn Command.Common.alphaWarningMessage
+    Left err ->
+      throwProjectCreationError $
+        "Failed to parse absolute path to wasp project dir: " ++ show err
+
+-- To avoid creating the initial files for a new project dynamically,
```

Comment:
What does it mean "dynamically" in this context? I think it is a bit too complex to understand, since we also don't use this word in any other places in this way, I think. Can you simplify this whole comment -> just say smth like "Copies prepared files to set up the project" or smth like that. If needed you can add "does no template rendering/compilation", but maybe best to even avoid that.